### PR TITLE
Semi-Realtime MIDI for 2.1

### PR DESCRIPTION
**Pull request by shoogle**
Backporting my GSoC project for MuseScore 2.1

**Bugs inherited from master:**
- Problem switching between modes **[fixed]**
- New note grouping also affects step-time mode **[fixed]**
- Problems with real-time durations that don't fit beats **[fixed]**
- Problems with real-time durations that cross barlines **[fixed]**
- Problems combining overlapping notes **[fixed]**
- Ties added by grouping function are not "real" until score is saved and reloaded **[fixed]**

**New bugs in 2.1:**
- Function to regroup rhythms doesn't always work. **[fixed]**

### CODE REVIEW

File: mscore/shortcut.cpp [hunk 1]
```diff
@@ -640,6 +703,17 @@ Shortcut Shortcut::_sc[] = {
       {
          MsWidget::SCORE_TAB,
          STATE_NORMAL | STATE_NOTE_ENTRY,
+         "reset-groupings",
+         QT_TRANSLATE_NOOP("action","Reset Note and Rest Groupings"),
+         QT_TRANSLATE_NOOP("action","Reset note and rest groupings"),
```
Comment (IsaacWeiss) [inline mscore/shortcut.cpp]: https://github.com/IsaacWeiss/MuseScore/blob/791b5767a9948f2acc6a9859874f60b853bc4c78/mscore/shortcut.cpp ?

Comment (shoogle) [inline mscore/shortcut.cpp]: Thanks. I will change it to "Regroup rhythms" or whatever it is in the master branch.

**Comment by shoogle** [comment]
Ready to merge.

**Comment by lasconic** [comment]
OK. The only problem I see is that sometimes real time note entry doesn't work anymore. The root of the problem seems to be that "chord" is true here https://github.com/musescore/MuseScore/pull/3035/files#diff-39acd419478f3f0d33e494c8b37942b0R5202 and it's due to `active > 0` in MuseScore::midiNoteReceived(). 

This is a recurring problem even before this PR. People were blocked in chord mode, the cursor was not moving forward. This problem will hurt us even more with this PR since we will push more people to use MIDI keyboard with these features. I have no idea yet what causes active being > 0 when no notes are pressed but we could add a MIDI panic button and react on allNoteOff controller accordingly, that should decrement `active` to 0 and we will be good to go.

**Comment by lasconic** [comment]
The problem above is not directly linked to the PR, so let's merge and get some feedback.

**Pull request merged**